library/spdm_secured_message_lib/libspdm_secmes_encode_decode.c

```diff
@@ -127,8 +127,15 @@ libspdm_return_t libspdm_encode_secured_message(
         return LIBSPDM_STATUS_SEQUENCE_NUMBER_OVERFLOW;
     }
 
+    /* Form the AEAD IV from the salt and the sequence number. */
     libspdm_copy_mem (iv, LIBSPDM_MAX_AEAD_IV_SIZE, salt, aead_iv_size);
-    data64 = libspdm_read_uint64((const uint8_t *)iv) ^ sequence_number;
+
+    if (secured_message_context->sequence_number_endian ==
+        LIBSPDM_DATA_SESSION_SEQUENCE_NUMBER_ENDIAN_BIG) {
+        data64 = libspdm_read_uint64((const uint8_t *)iv) ^ libspdm_le_to_be_64(sequence_number);
```

Comment:
If the intention is to simulate what TLS does.

Then I think we should XOR IV's most significant bit, not least significant bit.

See the picture at https://github.com/DMTF/libspdm/issues/2166#issuecomment-1608838324

Reply:
Yes, that needs to be fixed.